summingbird-online/src/main/scala/com/twitter/summingbird/online/MergeableStoreFactory.scala

```diff
@@ -26,23 +27,26 @@ import com.twitter.summingbird.batch.{ Batcher, BatchID }
  */
 object MergeableStoreFactory {
 
-  def apply[K, V](store: () => Mergeable[K, V], batcher: Batcher) = {
+  def apply[K, V](store: () => Mergeable[K, V], batcher: Batcher) =
+    fromWithSemigroup { _: Semigroup[V] => store() }(batcher)
+
+  def from[K, V](store: => Mergeable[(K, BatchID), V])(implicit batcher: Batcher): MergeableStoreFactory[(K, BatchID), V] =
+    fromWithSemigroup(_ => store)
+
+  def fromWithSemigroup[K, V](fn: Semigroup[V] => Mergeable[K, V])(implicit batcher: Batcher): MergeableStoreFactory[K, V] =
     new MergeableStoreFactory[K, V] {
-      def mergeableStore = store
+      def mergeableStore = fn
       def mergeableBatcher = batcher
     }
-  }
 
-  def from[K, V](store: => Mergeable[(K, BatchID), V])(implicit batcher: Batcher): MergeableStoreFactory[(K, BatchID), V] =
-    apply({ () => store }, batcher)
+  def fromOnlineOnly[K, V](store: => MergeableStore[K, V]): MergeableStoreFactory[(K, BatchID), V] =
+    fromOnlineOnlyWithSemigroup(_ => store)
 
-  def fromOnlineOnly[K, V](store: => MergeableStore[K, V]): MergeableStoreFactory[(K, BatchID), V] = {
-    implicit val batcher = Batcher.unit
-    from(store.convert { k: (K, BatchID) => k._1 })
-  }
+  def fromOnlineOnlyWithSemigroup[K, V](fn: Semigroup[V] => MergeableStore[K, V]): MergeableStoreFactory[(K, BatchID), V] =
+    fromWithSemigroup(fn.andThen(_.convert { k: (K, BatchID) => k._1 }))(Batcher.unit)
 }
 
 trait MergeableStoreFactory[-K, V] extends java.io.Serializable {
-  def mergeableStore: () => Mergeable[K, V]
+  def mergeableStore: Semigroup[V] => Mergeable[K, V]
```

Comment:
I didn't mean to suggest we change the structure, just the name. MergeableStoreFactory.mergeableStore reads like a property which would provide a mergeable store but it is a function that takes an input to provide a mergeable store. 

e.g. when I read factory.mergeableStore I would think that this is a mergeable store object, finding out that it is a function that needs a semigroup as input to produce a mergeable store would be a surprise.